Create st_tr_rt.rb

Asdfif opened the pull request:
Класс Station (Станция)
Класс Route (Маршрут)
Класс Train (Поезд)

Asdfif committed:
Create st_tr_rt.rb

russo-matrosso commented on st_tr_rt.rb:
```diff
@@ -0,0 +1,153 @@
+
+	attr_accessor :trains , :name, :type
+	def initialize(name)
+		@name = name
```
Отступы строго в два пробела, настрой на это редактор

russo-matrosso commented on st_tr_rt.rb:
```diff
@@ -0,0 +1,153 @@
+class Station
+
```
https://github.com/arbox/ruby-style-guide/blob/master/README-ruRU.md#empty-lines-around-bodies
Вообще, весь стайлгайд прочитать и использовать

russo-matrosso commented on st_tr_rt.rb:
```diff
@@ -0,0 +1,153 @@
+	end
+
+	def show_all_trains
+		@trains.each { |train| puts train.name}
```
puts тут не нужны. Объекты общаются при помощи данных, а не выводами в терминал

russo-matrosso commented on st_tr_rt.rb:
```diff
@@ -0,0 +1,153 @@
+	end
+
+	def show_type_trains(type)
+		@trains.each { |train| 
```
Если блок занимает несколько строк - он записывается при помощи do-end, а не {}

russo-matrosso commented on st_tr_rt.rb:
```diff
@@ -0,0 +1,153 @@
+		puts ''
+	end	
+
+	def send_train(train)
```
Станция не должна руководить отправкой поезда. У нее должен быть только метод для удаления поезда из списка

russo-matrosso commented on st_tr_rt.rb:
```diff
@@ -0,0 +1,153 @@
+
+	def send_train(train)
+		puts "Куда отправить поезд? f/b?"
+		q = gets.chomp
```
Что значит переменная q? Почему ее нельзя назвать, например, direction?

russo-matrosso commented on st_tr_rt.rb:
```diff
@@ -0,0 +1,153 @@
+
+	attr_accessor  :stations
+	def initialize(first, terminal)
+		if first.class == Station && terminal.class == Station
```
Если передается не Station - это ошибка, программа должна упасть. Иначе потом словим кучу ошибок с nil

russo-matrosso commented on st_tr_rt.rb:
```diff
@@ -0,0 +1,153 @@
+
+	def add_station(station)
+		unless @stations.include?(*station)
+		@stations.insert(-2, *station)
```
Отступы.

russo-matrosso commented on st_tr_rt.rb:
```diff
@@ -0,0 +1,153 @@
+				if station != @stations[0] && station != @stations[-1]
+				@stations.delete(station)
+				else
+					puts 'Нельзя удалить терминальные станции маршрута' 
```
Выкидываем все puts. Сейчас какая-то смесь интерфейса и данных

russo-matrosso commented on st_tr_rt.rb:
```diff
@@ -0,0 +1,153 @@
+		@type = type
+		@vagons = vagons
+		@current_speed = 0
+		@route = nil
```
Не нужно назначать пустым переменным экземпляра nil. Они равны ему по-умолчанию

russo-matrosso commented on st_tr_rt.rb:
```diff
@@ -0,0 +1,153 @@
+		puts "Текущая скорость - #{@current_speed}"
+	end
+	
+	def vagons; puts " Количество вагонов - #{@vagons}"; end
```
Зачем тут метод в одну строку?

russo-matrosso commented on st_tr_rt.rb:
```diff
@@ -0,0 +1,153 @@
+		@route = route
+		@current_station = route.stations[0]
+		@route.stations[0].trains << self
+		@route.stations.each { |station| puts station.name}
```
Мы назначаем маршрут и выводим названия станций. Зачем? Метод должен делать только то, о чем просят

russo-matrosso commented on st_tr_rt.rb:
```diff
@@ -0,0 +1,153 @@
+	end
+
+	def forward 
+		i = @route.stations.index(@current_station)
```
Лучше завести методы для нахождения следующей и предыдущей станции и использовать их

Asdfif committed:
Внесены поправки к заданию

Asdfif committed:
Update st_tr_rt.rb

Внесены поправки к заданию (2): поправки от рубокопа,  добавлены методы нахождения следующей и предыдущей станций, изменены методы перемещения поезда вперед и назад.

russo-matrosso commented on st_tr_rt.rb:
```diff
@@ -1,122 +1,110 @@
+
+  def faster(speed)
+    @current_speed += speed
+    @current_speed
```
Эта строка необязательна. При сложении вернется результат сложения

russo-matrosso commented on st_tr_rt.rb:
```diff
@@ -1,122 +1,110 @@
+  def slower(speed)
+    @current_speed -= speed
+    @current_speed = 0 if @current_speed <= 0
+    @current_speed
```
Тоже необязательная строка

russo-matrosso commented on st_tr_rt.rb:
```diff
@@ -1,122 +1,110 @@
+    @current_speed
+  end
+
+  attr_reader :vagons, :current_station
```
Почему-то ридер здесь, а не в начале класса

russo-matrosso commented on st_tr_rt.rb:
```diff
@@ -1,122 +1,110 @@
+    @vagons -= 1 if @current_speed.zero? && @vagons >= 1
+  end
+
+  def speed
```
Есть же аксессор на current_speed

russo-matrosso commented on st_tr_rt.rb:
```diff
@@ -1,122 +1,110 @@
+    return unless next_station
+
+    i = @route.stations.index(@current_station)
+    @current_station = @route.stations[i += 1]
```
Используй метод previous_station
`@current_station = previoust_station` итд

russo-matrosso replied:
Иначе получается дублирование кода

Asdfif committed:
Внесены изменения 07.11

Убрал необязательные строки. изменил методы перемещения поезда

russo-matrosso commented on st_tr_rt.rb:
```diff
@@ -0,0 +1,103 @@
+
+  def show_type_trains(type)
+    @trains.each do |train|
+      train if train.type == type
```
чуть лучше
`trains.filter { |train| train.type == type }`
